ocaml/libs/tracing/tracing_export.ml

```diff
@@ -0,0 +1,346 @@
+(*
+ * Copyright (C) 2024 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module D = Debug.Make (struct let name = "tracing-export" end)
+
+module Delay = Xapi_stdext_threads.Threadext.Delay
+open D
+open Tracing
+
+let ( let@ ) f x = f x
+
+module Export = struct
+  let export_interval = ref 30.
+
+  let set_export_interval t = export_interval := t
+
+  let host_id = ref "localhost"
+
+  let set_host_id id = host_id := id
+
+  let service_name = ref None
+
+  let set_service_name name = service_name := Some name
+
+  let get_service_name () =
+    match !service_name with
+    | None ->
+        warn "service name not yet set!" ;
+        "unknown"
```

Comment:
Is this worth the effort when we are using a default? Why not `let service_name = ref "unnkown"` and don't deal with options?